### PR TITLE
Add Cambium ePMP 1000 Login Scanner module

### PR DESCRIPTION
**Pull request by juushya**
This module scans for Cambium ePMP 1000 management login portal(s), and attempts to identify valid credentials. Default login credentials are - admin/admin, installer/installer, home/home and readonly/readonly.

## Verification

```
msf > use auxiliary/scanner/http/epmp1000_web_login
msf auxiliary(epmp1000_web_login) > info

       Name: Cambium ePMP 1000 Login Scanner
     Module: auxiliary/scanner/http/epmp1000_web_login
    License: Metasploit Framework License (BSD)
       Rank: Normal

Provided by:
  Karn Ganeshen <KarnGaneshen@gmail.com>

Basic options:
  Name              Current Setting  Required  Description
  ----              ---------------  --------  -----------
  BLANK_PASSWORDS   false            no        Try blank passwords for all users
  BRUTEFORCE_SPEED  5                yes       How fast to bruteforce, from 0 to 5
  DB_ALL_CREDS      false            no        Try each user/password couple stored in the current database
  DB_ALL_PASS       false            no        Add all passwords in the current database to the list
  DB_ALL_USERS      false            no        Add all users in the current database to the list
  PASSWORD          admin            no        A specific password to authenticate with
  PASS_FILE                          no        File containing passwords, one per line
  Proxies                            no        A proxy chain of format type:host:port[,type:host:port][...]
  RHOSTS                             yes       The target address range or CIDR identifier
  RPORT             80               yes       The target port
  SSL               false            no        Negotiate SSL/TLS for outgoing connections
  STOP_ON_SUCCESS   false            yes       Stop guessing when a credential works for a host
  THREADS           1                yes       The number of concurrent threads
  USERNAME          admin            no        A specific username to authenticate as
  USERPASS_FILE                      no        File containing users and passwords separated by space, one pair per line
  USER_AS_PASS      false            no        Try the username as the password for all users
  USER_FILE                          no        File containing usernames, one per line
  VERBOSE           true             yes       Whether to print output for all attempts
  VHOST                              no        HTTP server virtual host

Description:
  This module scans for Cambium ePMP 1000 management login portal(s), 
  and attempts to identify valid credentials. Default login 
  credentials are - admin/admin, installer/installer, home/home and 
  readonly/readonly.

msf auxiliary(epmp1000_web_login) > set rhosts 10.10.10.20
msf auxiliary(epmp1000_web_login) > set username installer
msf auxiliary(epmp1000_web_login) > set password installer
msf auxiliary(epmp1000_web_login) > run

[+] 10.10.10.20:80 - Running Cambium ePMP 1000 version 3.0...
[*] 10.10.10.20:80 - Trying username:"installer" with password:"installer"
[+] SUCCESSFUL LOGIN - 10.10.10.20:80 - "installer":"installer"
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

**Note**: msftidy run recommends using `res.get_cookies` instead of reading `Set-Cookie` via `res.headers`. However, the challenge with this specific target is the cookie has two (out of total 3) components that the modules needs to work with. The Cookie is of the form: `param1=val1;param2=val2;param3=val3;`

`res.get_cookies` seems to only pick up `param1`. Thus, I have used `res.headers['Set-Cookie']` in the module. And therefore, `msftidy` check is failing.

Is there a way to use `res.get_cookies` to pick up param3 as well?

Thank You.

### CODE REVIEW

File: modules/auxiliary/scanner/http/epmp1000_web_login.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
+## Final copy
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'cgi'
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/http/epmp1000_web_login.rb]: What's this for?

**Comment by wchen-r7** [comment]
Hi @juushya Could you please run msftidy on your code and fix all the warnings/errors? Thanks!

File: modules/auxiliary/scanner/http/epmp1000_web_login.rb [hunk 2]
```diff
@@ -0,0 +1,188 @@
+## Final copy
```
Comment (juushya) [inline modules/auxiliary/scanner/http/epmp1000_web_login.rb]: Done. Thanks.

File: modules/auxiliary/scanner/http/epmp1000_web_login.rb (hunk 1 above)
Comment (juushya) [inline modules/auxiliary/scanner/http/epmp1000_web_login.rb]: Not needed. Removed. Thanks.

**Comment by juushya** [comment]
Hi @wchen-r7 There is an issue with using res.get_cookies in that it seems to only capture the first of the cookie parameters. Can you please check my note after the module verification snippet above, and suggest any pointers? Thanks.

**Comment by wvu-r7** [comment]
What does the actual cookie look like? I can get all params with ```get_cookies```:

```
>> res = active_module.send_request_cgi; nil
=> nil
>> res.headers['Set-Cookie'] = 'param1=val1;param2=val2;param3=val3;'
=> "param1=val1;param2=val2;param3=val3;"
>> res.get_cookies
=> "param1=val1; param2=val2; param3=val3;"
>> 
```

**Comment by juushya** [comment]
Hi @wvu-r7 the cookie is of the form:
`sysauth=value1; path=value2;stok=value3`

`res.get_cookies` return `sysauth=value; `

**Note**: There is a space between `value1;` and `path`.

**Comment by wvu-r7** [comment]
@juushya: ```get_cookies``` skips ```path```, and it expects a ```;``` at the end of every key-value pair. IMHO, it's totally fine to use ```res.headers['Set-Cookie']``` if ```get_cookies``` is broken. We can fix ```get_cookies```.

**Comment by juushya** [comment]
Thanks @wvu-r7. I will leave it as is then. Is there any other info needed on this module from my end..?

File: modules/auxiliary/scanner/http/epmp1000_web_login.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
+##
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/http/epmp1000_web_login.rb]: Remove this line.

**Comment by wvu-r7** [comment]
@juushya: Have you seen [CONTRIBUTING.md](https://github.com/rapid7/metasploit-framework/blob/master/CONTRIBUTING.md)? Just follow guidelines from that so we can skip most of the code review.

**Comment by juushya** [comment]
# Release Notes

This login scanner module scans for Cambium ePMP 1000 management login portal(s), and attempts to identify version number & valid credentials.

**Comment by juushya** [comment]
@wvu-r7 msftidy + rubocop checks are clean + doc added. Travis checks fail on Set-Cookie usage only.

**Comment by wwebb-r7** [comment]
@juushya since no one has picked this up, I'm assuming that we don't have access to the hardware at this time.  If you could provide a pcap of a scan against both a Cambium device and something else (to show the negative case), I'll see about getting this landed.

**Comment by juushya** [comment]
Sure @wwebb-r7 Will send pcaps for Cambium modules tonight. Thanks.

**Comment by wwebb-r7** [comment]
It might help if I gave you an address ....
Send it to msfdev@metasploit.com

Sorry about that.  If you already emailed it elsewhere, I'll ask around.

**Comment by juushya** [comment]
@wwebb-r7 I have sent the pcaps for PR #7784, #7897, #8050 now. Please take a look. Thanks.

**Comment by wwebb-r7** [comment]
Got them, and all of them look good, thanks.  I don't think the CI error will be an issue, but I'll double check with a few people soon.  After that's confirmed, it should be merged fairly quick.

**Comment by wvu-r7** [comment]
@wwebb-r7: I think it's fine. I'll fix `get_cookies` as per https://github.com/rapid7/metasploit-framework/pull/7784#issuecomment-270998701.

**Comment by wwebb-r7** [comment]
### Release Notes

This update adds a module which scans for and identifies Cambium ePMP 1000 management login portals

**Pull request merged**